### DISCUSSION TITLE
Checkpoints and Restarts Using Vector Variables

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#21998] [Q&A General] [answered]
I am implementing the use of restart from a known steady state solution. I have run into a problem with loading the variable velocity which in my app is defined as a vector variable. My new solution will require the use of the velocity defined as vector variable as well, since the kernels are setup to use the variable velocity. Upon using the restart method defined in a previous message listed below I find that the variable listed in the solutions .e file are for each component of velocity and not velocity itself. 

How do I formulate my argument to read in the velocity components without having to redefine them in the kernels that now currently use the velocity vector function? I appear to be stuck in a do-loop for doing a restart.

"you may use either the checkpoint or exodus restart systems to achieve this.
The checkpoint system requires you to keep the exact same variables between both simulations. The exodus one is a little more lenient but you ll have to be careful to restart everything you need
https://mooseframework.inl.gov/application_usage/restart_recover.html"

**Comment by GiudGiud** [comment, marked as answer]
Checkpoint would be more appropriate for a vector variable. I havent done restart with vector variables.

You could probably make the exodus restart work going through these two objects
https://mooseframework.inl.gov/source/ics/VectorFunctionIC.html
https://mooseframework.inl.gov/source/functions/SolutionFunction.html

but I've never done it either

**Reply by TLWise** [reply]
Since the pressure drop establishes the velocity, maybe an initial solution for pressure distribution will provide a good start point for the velocity determination? I'm just trying to reduce the number of iterations needed to get to an initial transient solution from the established steady state solution.

**Reply by GiudGiud** [reply]
Typically you ll want to initialize both velocity and pressure when you can.
If you are running into issues with that, you can ramp up the boundary conditions or ramp down the viscsity over a few seconds of the transient to initialize
